src/test/java/tests/block/BlockListTests.java

```diff
@@ -0,0 +1,103 @@
+package tests.block;
+
+import base.BaseTest;
+import io.qameta.allure.Epic;
+import io.qameta.allure.Feature;
+import microservices.block.models.BlockListModel;
+import microservices.block.models.BlockModels;
+import microservices.block.steps.BlockSteps;
+import org.apache.commons.collections.MultiMap;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import util.CreateMultiParameters;
+
+import java.net.HttpURLConnection;
+
+@Epic("cardano")
+@Feature("api-block")
+public class BlockListTests extends BaseTest {
+    BlockSteps blockSteps = new BlockSteps();
+    private String blockId = "950010";
+
+    @Test(description = "get block list successfully", groups = {"block","block-list"})
+    public void getBlockListSuccessfully(){
+        BlockListModel blockListModel = (BlockListModel) blockSteps
+                .getTxListOfBlock(blockId)
+                .validateStatusCode(HttpURLConnection.HTTP_OK)
+                .saveResponseObject(BlockListModel.class);
+
+        blockSteps.then_verifyDataNotNull(blockListModel);
+
+        String blockId = "950073";
+        blockListModel = (BlockListModel) blockSteps
+                .getTxListOfBlock(blockId)
+                .validateStatusCode(HttpURLConnection.HTTP_OK)
+                .saveResponseObject(BlockListModel.class);
+
+        blockSteps.then_verifyDataNull(blockListModel);
+    }
+    @Test(description = "get block list unsuccessfully with invalid page", groups = {"block","block-list"}, dataProvider = "paramInvalidPage")
+    public void getBlockListUnsuccessfullyWithInvalidPage(Object page){
+        MultiMap param = new CreateMultiParameters()
+                .withPage(page)
+                .getParamsMap();
+        BlockListModel blockListModel = (BlockListModel) blockSteps
+                .getTxListOfBlock(blockId, param)
+                .validateStatusCode(HttpURLConnection.HTTP_OK)
+                .saveResponseObject(BlockListModel.class);
+
+        blockSteps.then_verifyDataNotNull(blockListModel);
+
+    }
+    @DataProvider(name ="paramInvalidPage")
+    public Object[][] dataSetInvalidPage(){
+        return new Object[][]{
+                {"abc"},
+                {"-10"},
+                {" "},
+                {"@#$%%"},
+        };
+    }
+    @Test(description = "get block list unsuccessfully with invalid blockId", groups = {"block","block-list"}, dataProvider = "paramInvalidBlockId")
+    public void getBlockListUnsuccessfullyWithInvalidBlockId(Object blockId){
```

Comment:
and here

Reply:
this function is going to give null data so i think it should name unsuccess